src/test/custom_node_buffersqueue_test.cpp

```diff
@@ -0,0 +1,142 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <cstring>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../custom_nodes/common/buffersqueue.hpp"
+
+using namespace ovms;
+using custom_nodes_common::BuffersQueue;
+
+TEST(CustomNodeBuffersQueue, GetAllBuffers) {
+    const std::string content{"abc"};
+    size_t buffersCount = 12;
+    BuffersQueue buffersQueue(content.size(), buffersCount);
+    std::vector<void*> buffers(buffersCount);
+    for (size_t i = 0; i < buffersCount; ++i) {
+        buffers[i] = buffersQueue.getBuffer();
+        ASSERT_NE(nullptr, buffers[i]) << "Failed to get: " << i;
+        std::memcpy(buffers[i], content.c_str(), content.size());
+    }
+    for (size_t i = 0; i < buffersCount; ++i) {
+        EXPECT_EQ(0, std::memcmp(buffers[i], content.c_str(), content.size()))
+            << "Buffer: " << i << " has different content";
+    }
+    std::sort(buffers.begin(), buffers.end());
```

Comment:
Do we sort the buffers in reverse order? If so, there should be comment specifying that, it might not be clear.

Reply:
Answer in previous comment response